src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,753 @@
+</span>
+<span class="language-py">
+  [pipeline options]({{ site.baseurl }}/documentation/sdks/pydoc/{{ site.release_latest }}/apache_beam.options.pipeline_options.html#apache_beam.options.pipeline_options.GoogleCloudOptions).
+</span>
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryTableSpecWithoutProject
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_table_spec_without_project
+%}```
+
+#### Using a TableReference
+
+To specify the target table with a `TableReference`, create a new
+`TableReference` using the three parts of the BigQuery table name.
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryTableSpecObject
+%}```
+```py
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/sdks/python/apache_beam/examples/snippets/snippets.py tag:model_bigqueryio_table_spec_object
+%}```
+
+<!-- Java specific -->
+
+{:.language-java}
+The Beam SDK for Java also provides the `parseTableSpec` helper method, which
+constructs a `TableReference` object from a String that contains the
+fully-qualified BigQuery table name. However, the static factory methods for
+BigQueryIO transforms accept the table name as a String and construct a
+`TableReference` object for you.
+
+
+### Table rows
+
+{:.language-java}
+BigQueryIO read and write transforms produce and consume data as either a
+`PCollection` of BigQuery `TableRow` objects, or a `PCollection` of user-defined
+types.
+
+{:.language-py}
+BigQueryIO read and write transforms produce and consume data as a `PCollection`
+of dictionaries, where each element in the `PCollection` represents a single row
+in the table.
+
+### Schemas
+
+When writing to BigQuery, you must supply a table schema for the fields you want
+to write to the target table unless you specify a [create disposition]() of
+`CREATE_NEVER`. [Creating a table schema](#creating-a-table-schema) covers
+schemas in more detail.
+
+
+## Reading from BigQuery
+
+BigQueryIO allows you to read from a BigQuery table, or read specific fields
+using a query string. When you apply a BigQueryIO read transform in batch mode,
+Beam invokes a [BigQuery export request](https://cloud.google.com/bigquery/docs/exporting-data).
+Beam’s use of this API is subject to BigQuery's [Quota](https://cloud.google.com/bigquery/quota-policy#export)
+and [Pricing](https://cloud.google.com/bigquery/pricing) policies.
+
+
+<!-- Java specific -->
+
+{:.language-java}
+The Beam SDK for Java has two BigQueryIO read methods. Both of these methods
+allow you to read from a table, or read fields using a query string.
+
+{:.language-java}
+1. `readTableRows` returns a `PCollection` of BigQuery `TableRow`
```

Comment:
I'd recommend to swap the order of these two methods because the second is typically preferable.

Reply:
Done